genesis/recorders/plotters.py

```diff
@@ -93,13 +125,13 @@ def build(self):
 
         if self._options.labels is not None:
             self._setup_plot_structure(self._options.labels)
+        else:
+            self._setup_plot_structure(self._data_func())
```

Comment:
I'm not a fan of this logics. I would rather have `_setup_plot_structure` taking two input arguments, with data being specified systematically, which is good to implement some asserts checking consistency between labels and data at build-time.
```python
self._setup_plot_structure(self._options.labels, self._data_func())
```